azurelinuxagent/ga/exthandlers.py

```diff
@@ -371,11 +371,14 @@ def _extension_processing_allowed(self):
                 logger.info("Extension handling is on hold")
                 return False
 
+        if self.last_etag == etag:
```

Comment:
Let's move this check to the if statement at line 281 or think of a better name for '_extension_processing_allowed' to reflect better what it is doing.

Reply:
I think the name is pretty apt. The name `_extension_processing_allowed` implies that we only proceed if its allowed and the incarnation check is one of those checks that we use to determine if its allowed.
How about a docstring? 
Or maybe renaming it to `_extension_processing_allowed_if_incarnation_changed` maybe?

Reply:
no, i don't think the name is pretty apt... the check for the incarnation seems hidden in this function. _extension_processing_allowed_if_incarnation_changed doesn't seem apt either. I'd move the check one level up

Reply:
Done